packages/x-data-grid/src/components/columnHeaders/GridColumnHeaderSeparator.tsx

```diff
@@ -61,7 +61,7 @@ function GridColumnHeaderSeparatorRaw(props: GridColumnHeaderSeparatorProps) {
     // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions
     <div
       className={classes.root}
-      style={{ minHeight: height, opacity: rootProps.showColumnVerticalBorder ? 0 : 1 }}
+      style={{ minHeight: height, opacity: rootProps.showColumnVerticalBorder ? 0 : undefined }}
```

Comment:
I am observing the same behaviour on `master`. 

Is this prop supposed to be the equivalent of `showCellVerticalBorder`, but for the column header item? `showCellVerticalBorder` adds the border to both cells and column headers currently.

Reply:
👍 Correct, the behavior is the same on the `master` as well. Feel free to move it out of the scope of this PR.

> Is this prop supposed to be the equivalent of showCellVerticalBorder, but for the column header item? showCellVerticalBorder adds the border to both cells and column headers currently.

I think so. That's what a user would expect from:
https://github.com/mui/mui-x/blob/97afd65d2338b42982923771dc647e20e4d6676a/packages/x-data-grid/src/models/props/DataGridProps.ts#L301-L310

Looks like we need to update [this](https://github.com/mui/mui-x/blob/97afd65d2338b42982923771dc647e20e4d6676a/packages/x-data-grid/src/components/columnHeaders/GridColumnHeaderItem.tsx#L135) and [this](https://github.com/mui/mui-x/blob/97afd65d2338b42982923771dc647e20e4d6676a/packages/x-data-grid/src/components/columnHeaders/GridColumnGroupHeader.tsx#L136) line to use `showColumnVerticalBorder` instead.

I'm not sure though if it will keep showing the separator properly too.

Reply:
Thanks @MBilalShafi - made the fix here as it's very much related.